plutus-core/plutus-ir/test/recursion/factorial.golden

```diff
@@ -15,7 +15,7 @@
                   [
                     [
                       (force (builtin ifThenElse))
-                      [ [ (builtin equalsInteger) x_514 ] (con integer 0) ]
+                      [ [ (builtin equalsInteger) (con integer 0) ] x_514 ]
```

Comment:
It's cool that it caught this but I'm not sure how. Looking at the code, I don't think that this test should have gone through the simplifier.